app/models/user.rb

```diff
@@ -410,6 +410,7 @@ def approve(approved_by, send_mail = true)
 
     if approved_by.is_a?(Integer)
       self.approved_by_id = approved_by
+      self.approved_by = User.find(approved_by)
```

Comment:
Hmm I'm not sure if this line is required here. Since the bug is due to using a variable that is named the same as a column in the table, perhaps the variable should be renamed? Maybe `approver`?

Reply:
> Since the bug is due to using a variable that is named the same as a column in the table

Yes, that's a good point.

Let's take a step back, and look at why the argument `approved_by` is not ALWAYS a User object.

![discourse____discourse_discourse__-_____app_models_user_rb__discourse_](https://user-images.githubusercontent.com/45508821/54201665-21099400-44c6-11e9-9f05-3e367b6ddc48.png)

It is only because of the `invite_redeemer.rb`, I think we can:

1. Change the argument name to `approver` which should always be a User object
2. Clean this up a bit here.

But let's first fix this bug now, and I can refactor. WDYT?

Reply:
hehe I would just go for the refactor and remove the nightmare where a variable can be an id or an ActiveRecord object.

Reply:
😂 Let's do it. 🏃 🏃 🏃

Reply:
@tgxworld How about now? 👀